src/components/SignalGeneratorComponent.cpp

```diff
@@ -0,0 +1,90 @@
+/**
+ * @file SignalGeneratorComponent.cpp
+ * @author Luciana Moreno Borges (lumoreno.borges@gmail.com)
+ * @brief Signal Generator component
+ * @version 0.1
+ * @date 2019-01-25
+ * 
+ * @copyright Copyright (c) 2019
+ * 
+ */
+#include <hyro/SignalGeneratorComponent.h>
+#include <hyro/msgs/common/Basic.h>
+#include <hyro/SignalGenerator.h>
+#include <hyro/msgs/Signal.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+
+#include <iostream>
+
+namespace hyro
+{
+  SignalGeneratorComponent::SignalGeneratorComponent(URI uri): hyro::Component(uri){
+
+  }
+  /* Create a log variable */
+  std::shared_ptr<hyro::HyroLogger> SignalGeneratorComponent::s_logger = hyro::HyroLoggerManager::CreateLogger("SignalGeneratorComponent");
+  
+  /*Initialize the object output and register the dynamic properties*/
+  hyro::Result
+  SignalGeneratorComponent::init (const hyro::ComponentConfiguration & config)
+  {
+    signal = SignalGenerator();
+    std::shared_ptr<ChannelOutput<std::vector<int>>>
+      m_dummy = this->registerOutput<std::vector<int>>("fix_dynamic"_uri, config);
+    s_logger->info("Init:");
+  
+    m_output = this->registerOutput<Signal>("signals"_uri, config);
+
+    m_amplitude = 5.0f;
+    m_frequency = 3.0f;
+    m_cosine = true;
+
+    registerDynamicProperty<float>("amplitude", m_amplitude, [this](const float value){return value>=0 && value<=5;});
+    registerDynamicProperty<float>("frequency", m_frequency, [this](const float value){return value>=0 && value<=5;});
+    registerDynamicProperty<bool>("cosine", m_cosine, [this](const bool value){return value==true || value==false;});
+    DynamicPropertyAccess dynamic_property_access("/signal_generator"_uri);
```

Comment:
Use `DynamicPropertyAccess`in the main function.

Reply:
@lumoreno17 ping

Reply:
You can remove lines 43 to 45 from here. You already placed them in `signal_generator`

Reply:
@joaobrittoneto I registered it on SignalGeneratorComponent.cpp and I acessed it on signal_generator.cpp. Don't I need to register it anymore ?

Reply:
@joaobrittoneto Sorry, Now I saw that you were talking about the Acess

Reply:
@Actually you asked me to remove the register, so I dind't get it